msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/mwpj/TestCase2521960.kt

```diff
@@ -0,0 +1,196 @@
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TestRule
+
+// https://identitydivision.visualstudio.com/Engineering/_workitems/edit/2521960
+// [MWPJ] Device registration entry migration (different upn - same tenant)
+@LocalBrokerHostDebugUiTest
+@SupportedBrokers(brokers = [BrokerHost::class])
+class TestCase2521960 : AbstractMsalBrokerTest() {
+
+    private lateinit var mUsGovAccount: ILabAccount
+    private lateinit var mLabAccount2: ILabAccount
+    private lateinit var mBrokerHostApp: BrokerHost
+
+    @get:Rule
+    val loadUsGovLabAccountUserRule: TestRule = LoadLabUserTestRule(getAdditionalLabQuery())
+    @get:Rule
+    val loadAdditionalLabUserRule: TestRule = LoadLabUserTestRule(TempUserType.BASIC)
+
+    @Test
+    fun test_2521960() {
+        // Register 2 accounts from different tenants
+        mBrokerHostApp.multipleWpjApiFragment.performDeviceRegistration(mUsGovAccount.username, mUsGovAccount.password)
+        mBrokerHostApp.multipleWpjApiFragment.performDeviceRegistration(mLabAccount.username, mLabAccount.password)
+        val deviceRegistrationRecords = mBrokerHostApp.multipleWpjApiFragment.allRecords
+        Assert.assertEquals(2, deviceRegistrationRecords.size)
+
+        // Unregister the device from the legacy space
+        mBrokerHostApp.multipleWpjApiFragment.unregister(mUsGovAccount.username)
+
+        // Verify that the device is unregistered for the legacy API
+        val errorMessage = mBrokerHostApp.accountUpn
+        Assert.assertNotNull(errorMessage)
+        Assert.assertTrue(errorMessage!!.contains("Device is not Workplace Joined"))
+
+        // Verify that the device is still registered for the second account using the MWPJ API.
+        val recordInExtendedSpace = mBrokerHostApp.multipleWpjApiFragment.getRecordByUpn(mLabAccount.username)
+        Assert.assertNotNull(recordInExtendedSpace)
+
+        // Register the device with the second account (same tenant different upn) using the legacy API
+        mBrokerHostApp.performDeviceRegistration(mLabAccount2.username, mLabAccount2.password)
+
+        // Verify that the device is registered with the second account using the legacy API
+        val legacyAccountMessage = mBrokerHostApp.accountUpn
+        Assert.assertNotNull(legacyAccountMessage)
+        Assert.assertTrue(legacyAccountMessage!!.contains(mLabAccount2.username))
+
+        // Verify the entry in the extended space was removed and replaced with the entry from the second account.
+        val recordInLegacy = mBrokerHostApp.multipleWpjApiFragment.allRecords
+        Assert.assertEquals(1, recordInLegacy.size)
+        Assert.assertEquals(recordInExtendedSpace["TenantId"], recordInLegacy[0]["TenantId"])
+        Assert.assertNotEquals(recordInExtendedSpace["Upn"], recordInLegacy[0]["Upn"])
+        Assert.assertNotEquals(recordInExtendedSpace["DeviceId"], recordInLegacy[0]["DeviceId"])
+
+        //  SSO shall not break (PRT is still usable without extra prompts)
```

Comment:
this step seems to be missing in the manual test suite.

Reply:
https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/1817/commits/86e143ba20948bf47febbc4e8f1df88ebc5e0ccd